### ISSUE TITLE
Memory corruption crashes without stack traces are not grouped together 

### ISSUE THREAD

**Issue by pbajurko** [bugsnag/bugsnag-android#904]
### Expected behavior
On our android client we see a lot of reports like that:
<img width="626" alt="SigSegv no stack trace" src="https://user-images.githubusercontent.com/65332883/87756180-b4647f00-c808-11ea-8171-c8cdc8d7f79a.png">

### Observed behavior
Every reported event is segmented as separated error because of different memory address. So we end up with thousands of 1 event errors, what doesn't help with assessing severity. 

### Steps to reproduce
We cannot reproduce issue locally atm. Though from breadcrumbs it happens after users close app so maybe some GC related issue.

### Version
5.0.0

### Additional information
1. I would like to understand how such event without any stacktrace can be produced.
2. I will try do some event processing on client side with custom hash grouping like described [here](https://docs.bugsnag.com/platforms/go/revel/customizing-error-reports/#overriding-bugsnag-39-s-grouping-algorithm). Any tips on error matching would be appreciated.

**Comment by pbajurko** [comment]
Hmm not sure about error class since we have other memory corruptions with full stack traces and those are grouped fine, so I don't wanna lump them together. 

I will write email with details thanks!

**Comment by pbajurko** [comment]
Following [example](https://github.com/bugsnag/bugsnag-android/blob/master/examples/sdk-app-example/src/main/cpp/entrypoint.cpp#L24) I added observer for native crashes and grouped them with such snippet

```
bool group_crashes_with_empty_stacktraces(void *event) {
    if (const char *errorClass = bugsnag_error_get_error_class(event)) {
        if (nullptr != strstr(errorClass, "SIGSEGV") &&
            bugsnag_event_get_stacktrace_size(event) < 3) {
            bugsnag_event_set_grouping_hash(event, "SIGSEGV without stacktrace");
        }
    }
    return true;
}
```

It works fine, thanks @johnkiely1  for all suggestions. Now if we can not figure out cause of those exceptions I suppose we may just [discard](https://docs.bugsnag.com/product/event-usage/#discard-individual-errors) them. If you guys don't think it is worth to improve grouping events with empty stack-traces on backend side I think we can close this issue.

**Comment by johnkiely1** [comment]
Hi pbajurko,

I'm glad you found a solution that works. I will close this for now but feel free to comment here if anything extra crops up and we can reopen as needs be.

**Comment by pbajurko** [comment]
While Bugsnag SDK does not catch any meaningful stack trace in this particular case, on Playstore crash dashboard we can find:

```
#00  pc 0000007b57492c84  <unknown>
#01  pc 000000000001ba58  /system/lib64/libc.so (__cxa_thread_finalize+40)
#02  pc 0000000000067e6c  /system/lib64/libc.so (pthread_exit+20)
#03  pc 0000000000067d60  /system/lib64/libc.so (__pthread_start(void*)+292)
#04  pc 000000000001f348  /system/lib64/libc.so (__start_thread+68)
```

Which seems to be related to [platform issue](https://github.com/android/ndk/issues/360) and calling dlclose before thread destructors . @johnkiely1 any idea why there is  discrepancy between Bugsnag and Playstore crash reports?

**Comment by johnkiely1** [comment]
Hey @pbajurko, going to close this as we have continued the discussion via email. 

For future travellers:

While we have not been able to reproduce the exact issue seen here, we have made significant improvements in the way we capture NDK stacktraces in [v5.21.0](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.21.0) of our Android notifier. We would advise upgrading to this version which should give more complete stack traces.
